integration-tests/tests/common/mod.rs

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! This crate contains the common code for integration tests
+
+#![deny(clippy::all)]
+#![deny(missing_docs)]
+
+use std::io::{Error, ErrorKind};
+use std::path::{Path, PathBuf};
+use std::process::Command;
+use std::time::Duration;
+use wait_timeout::ChildExt;
+use walkdir::WalkDir;
+
+pub fn keep_type() -> &'static str {
+    #[cfg(has_sgx)]
+    return "enarx-keep-sgx";
+
+    #[cfg(has_sev)]
+    return "enarx-keep-sev";
+
+    #[cfg(not(any(has_sgx, has_sev)))]
+    compile_error!("Need either SGX or SEV!");
+}
+
+pub struct IntegrationTest {
+    pub wksp_root: PathBuf,
+    pub keep: PathBuf,
+    pub payload: PathBuf,
+    pub shim: PathBuf,
+}
+
+impl IntegrationTest {
+    pub fn new(bin: &str) -> Self {
+        let enarx_dir = Path::new(env!("CARGO_MANIFEST_DIR"));
+        let enarx_dir = enarx_dir.parent().unwrap();
+        let keep_type = keep_type();
+
+        let wksp_root = enarx_dir.to_path_buf();
+        let keep_dir = enarx_dir
+            .join(format!("{}/target/", keep_type))
+            .to_path_buf();
+        let payload_dir = enarx_dir
+            .join("integration-tests/target/x86_64-unknown-linux-musl/")
+            .to_path_buf();
+        let shim_dir = enarx_dir
+            .join(format!(
+                "{}-shim/target/x86_64-unknown-linux-musl/",
+                keep_type
+            ))
+            .to_path_buf();
+        let shim_filename = format!("{}-shim", keep_type);
+        let shim_filename = shim_filename.as_str();
+
+        let keep = find_filepath(keep_dir, keep_type).unwrap();
+        let payload = find_filepath(payload_dir, bin).unwrap();
+        let shim = find_filepath(shim_dir, shim_filename).unwrap();
+
+        Self {
+            wksp_root,
+            keep,
+            payload,
+            shim,
+        }
+    }
+
+    pub fn run(self: Self, timeout: u64, exit_status: i32) -> () {
+        let seconds = Duration::from_secs(timeout);
+
+        let mut cmd = Command::new(self.keep)
+            .current_dir(self.wksp_root)
+            .arg("--code")
+            .arg(self.payload)
+            .arg("--shim")
+            .arg(self.shim)
+            .spawn()
+            .expect("failed to run the test");
+
+        let ecode = match cmd.wait_timeout(seconds).unwrap() {
+            Some(status) => status.code(),
+            None => {
+                cmd.kill().unwrap();
+                panic!("killed by watchdog!");
+            }
+        };
+
+        assert_eq!(exit_status, ecode.unwrap());
+    }
+}
+
+/// Finds an absolute file path for a file in a directory.
+#[cfg_attr(not(has_sgx), ignore)]
```

Comment:
This needs to be updated since `has_sgx` is not the only `cfg`.

I wonder if we should build on top of our `has_blah` `cfg`s so that we can more conveniently ignore code without repeating the list of `has_blah` `cfg`s.

I think something like:

```rust
#[cfg_attr(any(has_sgx, has_sev), has_tee)]
```

should work if put in the root of the crate. Either `integration-tests/src/main.rs` or maybe `integration-tests/tests/common/mod.rs`

Then it's more convenient because there are some functions like `keep_type()` that need to query for a specific `has_sgx` or `has_sev`, but there are other blocks of code that don't care _which_ TEE we're running under, just that there is one.

This will be especially nice as once we have even more TEEs we'll just need to remember to add their entries to `build.rs` and that `cfg_attr`